OmniKit/PumpManager/PodComms.swift

```diff
@@ -41,52 +49,147 @@ class PodComms: CustomDebugStringConvertible {
         self.messageLogger = nil
     }
     
-    private func assignAddress(address: UInt32, commandSession: CommandSession) throws -> PodState {
+    // Handles all the common work to send and verify the version response for the two pairing commands, AssignAddress and SetupPod
+    private func sendPairMessage(address: UInt32, transport: PodMessageTransport, message: Message) throws -> VersionResponse {
+
+        defer {
+            log.debug("sendPairMessage saving current transport packet #%d", transport.packetNumber)
+            if self.podState != nil {
+                self.podState!.messageTransportState = MessageTransportState(packetNumber: transport.packetNumber, messageNumber: transport.messageNumber)
+            } else {
+                self.startingPacketNumber = transport.packetNumber
+            }
+        }
+
+        var didRetry = false
+        var rssiRetries = numRssiRetries
+        while true {
```

Comment:
Not a fan of retrying at this level. It can be *slow* just to do one exchange, and it seems at this level that it's appropriate for the user to have more feedback, and decide what action to take next if a send message fails.

Reply:
After discussion, sounds like retrying should be minimal